### PR TITLE
Integrate update system

### PR DESCRIPTION
**Pull request by HirotoShioi**
This PR integrates the update system to the launcher executable. I've also added an document explaining how the update is being done, but it might be a better idea to move it over to wiki.

### How to test
Build and run `cardano-launcher`

To test that `runUpdater` works properly, I've prepared two bash script which will act as a mock of updater / daedalus.

https://github.com/input-output-hk/cardano-shell/blob/hiroto/integrate-update-system/test/testDaedalusFrontend.sh
https://github.com/input-output-hk/cardano-shell/blob/hiroto/integrate-update-system/test/testUpdater.sh

### CODE REVIEW

**Comment by erikd** [comment]
Seems many of the commits in this PR have already been merged. @HirotoShioi ?

**Comment by HirotoShioi** [comment]
@erikd I did not know this since I can still see `Squash and merge` button on my view so I've assumed it was still open.. Weird.

**Comment by ksaric** [comment]
@HirotoShioi You can rebase on `master` and have a clean picture afterward.

**Comment by erikd** [comment]
@HirotoShioi The problem is that you are doing "git merge master" or similar. You should do "git rebase master" instead.

File: app/Launcher/Main.hs [hunk 1]
```diff
@@ -36,9 +36,9 @@ import           Data.X509.Extra (failIfReasons, genRSA256KeyPair,
 -- Main
 --------------------------------------------------------------------------------
 
-main :: IO ()
+main :: IO ExitCode
```
Comment (erikd) [inline app/Launcher/Main.hs]: Errr, I really don't think that is right. The type of `main` should remain the same and you can then use `exitSuccess` or `exitFailure` to the the exectuable's exit code.

File: app/Launcher/Main.hs [hunk 2]
```diff
@@ -70,10 +70,21 @@ main = do
...
-    _ <- runWallet externalDependencies walletPath walletArgs
-
-    pure ()
+    runNode (runWallet externalDependencies walletPath walletArgs)
+  where
+    runNode :: IO ExitCode -> IO ExitCode
+    runNode runWalletFunc = do
+        eUpdateResult <- runUpdater testUpdaterData
+        case eUpdateResult of
+            Left _ -> return $ ExitFailure 1
```
Comment (erikd) [inline app/Launcher/Main.hs]: You can just replace that with `exitFailure 1`.

File: app/Launcher/Main.hs [hunk 3]
```diff
@@ -1,6 +1,5 @@
-{-# LANGUAGE LambdaCase      #-}
-{-# LANGUAGE RankNTypes      #-}
-{-# LANGUAGE RecordWildCards #-}
```
Comment (erikd) [inline app/Launcher/Main.hs]: :100:

File: app/Launcher/Main.hs [hunk 4]
```diff
@@ -79,7 +79,8 @@ main = do
 
     void $ runUpdater updaterData -- On windows, process dies here
     -- You still want to run the wallet even if the update fails
-    runWallet externalDependencies walletPath walletArgs updaterData
+    exitCode <- runWallet externalDependencies walletPath walletArgs updaterData
+    exitWith exitCode
```
Comment (erikd) [inline app/Launcher/Main.hs]: :+1:

**Pull request merged**